infrastructure/helm/quantumserverless/charts/gateway/templates/deployment.yaml

```diff
@@ -67,14 +67,24 @@ spec:
               value: {{ .Values.application.debug | quote }}
             - name: SITE_HOST
               value: {{ .Values.application.siteHost | quote }}
+            - name: RAY_HOST
+              value: {{ .Values.application.rayHost | quote }}
+            - name: SETTINGS_AUTH_MECHANISM
+              value: {{ .Values.application.auth.mechanism | quote }}
+            - name: SETTINGS_TOKEN_AUTH_URL
+              value: {{ .Values.application.auth.token.url | quote }}
+            - name: SETTINGS_TOKEN_AUTH_VERIFICATION_URL
+              value: {{ .Values.application.auth.token.verificationUrl | quote }}
+            - name: SETTINGS_TOKEN_AUTH_VERIFICATION_FIELD
+              value: {{ .Values.application.auth.token.verificationField | quote }}
```

Comment:
may need to wrap these in an `if platform == ibm` block

Reply:
mmm... Initially I would say that maybe you're right but it's a bit tricky because why should not we apply same logic for `keycloack`? Or even you want to use both methods? Thinking out loud, maybe we need to start reviewing our configuration.

Reply:
But as first approach to solve the problem maybe we can enable one method or another if we use or not `keycloack`, for example.

Reply:
keycloak uses different fields. for example, there's no `Values.application.auth.token` ... for keycloak it's `Values.application.auth.keycloak`. that's causing the non-ibm deploy to fail because none of the token values are set (so they're `nil` which throws an error). so we probably have to use the platform to select which value to set for these envvars, depending on the install platform (local / ibm / etc.)

Reply:
But I mean, it's the same for `token`, isn't it? There is no `Values.application.auth.keycloack` for `token`.

Reply:
I'd assume so... I only tested with a local install, but I'd assume the same thing would happen for missing keycloak values when deploying on IBM Cloud

Reply:
Weird, for me is working and locally there were never `token` values 🤔. Does this happen for you too, @akihikokuroda ?

Reply:
I got it to deploy successfully by adding `application.auth.token.url: "nil"` to `values.yaml`, so it seems like it's just that one field that's misbehaving...

Reply:
actually, the issue is the missing `token` ... that's what's getting rendered as `nil` and causing the failure

something like this would be totally fine:

```yaml
auth:
  keycloak:
    stuff...
  token: {}
```

(and probably vice-versa for the IBM config)

Reply:
Sounds good to me, I don't understand yet why in one way was happening but in another way not 🤷 . Thanks for solving it anyway, @psschwei !

Reply:
ok, I'm running around in circles here.... 

the default values for the gateway chart set the keycloak stuff, that's why those aren't failing on IBM Cloud 

if we set the token values in that chart (`helm/quantumserverless/charts/gateway/values.yaml`), then this all goes away and no need to add a blank `token: {}` to the main `values-ibm.yaml`

actually, maybe we should just change the gateway defaults to:

```yaml
application:
  keycloak: {}
  token: {}
```

and then set the real values when deploying to a specific environment...

Reply:
Solved with [2e320c0](https://github.com/Qiskit-Extensions/quantum-serverless/pull/668/commits/2e320c01c338b77f81e2cc4ffd9c57d14005ae62)!

Reply:
And solved last comment with [63eef02](https://github.com/Qiskit-Extensions/quantum-serverless/pull/668/commits/63eef020a174a570251118c40cc1c52d2046d73a), thank you for the good catch up @psschwei !